### ISSUE TITLE
Deprecate the "expression -> new_attribute" form of projection arguments

### ISSUE THREAD

**Issue by dimitri-yatsenko** [datajoint/datajoint-python#214]
Currently, we can rename attributes and new attributes using two notations
```python
result = rel.proj(cos = 'cos(ori)')
```
and 
```python
result = rel.proj('cos(ori)->cos')
```
The latter notation comes from the MATLAB version of DataJoint and I don't see it used often.  The code and the documentation will become simpler if we remove the second form from python. 

Any objections?

**Comment by fabiansinz** [comment]
Since there is a substantial difference in the syntax of DataJoint Matlab and Python already, I don't see why we need to keep this particular one. So, no objections.

**Comment by dimitri-yatsenko** [comment]
okay, the upcoming release will deprecate this feature.  All *args in projection and aggregation must be simple attribute names.

**Comment by dimitri-yatsenko** [comment]
one problem with `**kwargs` is that they come unordered.  The use has no control of the order of the attributes in the resulting projection.

**Comment by fabiansinz** [comment]
Based on the discussion we had on the `__getitem__` that does not matter.

**Comment by dimitri-yatsenko** [comment]
exactly. One more reason not to support numeric indexing of attributes.

**Comment by dimitri-yatsenko** [comment]
I have also quietly dropped the % notation for projections.  We used to be able to do this:
```python
rel % ('one', 'sum(two)->n')
```
as the equivalent of 
```python
rel.proj('one', n='sum(two)')
```
Only the latter remains now.

**Comment by fabiansinz** [comment]
I think nobody used that anyway.

**Comment by dimitri-yatsenko** [comment]
fixed in 4debbd88f
